packages/amazonq/src/app/inline/recommendationService.ts

```diff
@@ -0,0 +1,77 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+    InlineCompletionListWithReferences,
+    InlineCompletionWithReferencesParams,
+    inlineCompletionWithReferencesRequestType,
+} from '@aws/language-server-runtimes/protocol'
+import { CancellationToken, InlineCompletionContext, Position, TextDocument } from 'vscode'
+import { LanguageClient } from 'vscode-languageclient'
+import { SessionManager } from './sessionManager'
+
+export class RecommendationService {
+    static #instance: RecommendationService
+
+    public static get instance() {
```

Comment:
is it possible to avoid using singletons here? One of the problems in the current codewhisperer codebase is its pretty hard to navigate through a bunch of instances that may or may not have any state